tests/vmi_lifecycle_test.go

```diff
@@ -630,6 +631,61 @@ var _ = Describe("VMIlifecycle", func() {
 
 		})
 
+		Context("with node feature discovery", func() {
+
+			var options metav1.GetOptions
+			var cfgMap *k8sv1.ConfigMap
+			var originalData string
+
+			BeforeEach(func() {
+				options = metav1.GetOptions{}
+				cfgMap, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Get("kubevirt-config", options)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				originalData = cfgMap.Data[featuregates.FEATURE_GATES_KEY]
+				cfgMap.Data[featuregates.FEATURE_GATES_KEY] = featuregates.CPUNodeDiscoveryGate
+				_, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Update(cfgMap)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+			})
+
+			AfterEach(func() {
+				cfgMap, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Get("kubevirt-config", options)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+				cfgMap.Data[featuregates.FEATURE_GATES_KEY] = originalData
+				_, err = virtClient.CoreV1().ConfigMaps(namespaceKubevirt).Update(cfgMap)
+				ExpectWithOffset(1, err).ToNot(HaveOccurred())
+			})
+
+			It("the vmi with cpu.model matching a nfd label on a node should be scheduled", func() {
```

Comment:
We need a negative test too. The simplest thing which we do on other scheduling tests as well, is not adding the wanted label to a node but add a custom node selector on the vmi too. This way we would only allow node A but we can then check that it does not gescheduled because node A misses a required label.